### ISSUE TITLE
Syncing to testnet: Received an invalid Bitcoin Cash address as input.

### ISSUE THREAD

**Issue by christroutner** [simpleledger/SLPDB#31]
When trying to sync with testnet, my copy of SLPDB keeps crashing with this error:

```
[Query] getMintTransactions(a184e3715b124c6641da4eb0f2940f578c1f6a3525646b8d9a0b919b9b83898e)
STR =  [
  {
    "slp": null,
    "txid": "de66d78081b92a9913dcea43943ac2b82ec8962550fe62710e68d61ee2e03d05",
    "versionTypeHex": "01",
    "block": 1293084,
    "timestamp": "2019-03-16 15:08:28",
    "batonHex": "02",
    "quantityHex": "00000000000004d2"
  }
]

{ InvalidAddressError: Received an invalid Bitcoin Cash address as input.
    at new InvalidAddressError (/home/safeuser/SLPDB/node_modules/bchaddrjs-slp/src/bchaddr.js:518:15)
    at decodeAddress (/home/safeuser/SLPDB/node_modules/bchaddrjs-slp/src/bchaddr.js:177:9)
    at Object.toSlpAddress (/home/safeuser/SLPDB/node_modules/bchaddrjs-slp/src/bchaddr.js:133:17)
    at Function.Utils.toSlpAddress (/home/safeuser/SLPDB/node_modules/slpjs/lib/utils.js:37:24)
    at SlpTokenGraph.<anonymous> (/home/safeuser/SLPDB/SlpTokenGraph.js:296:46)
    at Generator.next (<anonymous>)
    at /home/safeuser/SLPDB/SlpTokenGraph.js:7:71
    at new Promise (<anonymous>)
    at __awaiter (/home/safeuser/SLPDB/SlpTokenGraph.js:3:12)
    at asyncForEach (/home/safeuser/SLPDB/SlpTokenGraph.js:285:77)
  name: 'InvalidAddressError',
  message: 'Received an invalid Bitcoin Cash address as input.',
  stack:
   'InvalidAddressError: Received an invalid Bitcoin Cash address as input.\n    at new InvalidAddressError (/home/safeuser/SLPDB/node_modules/bchaddrjs-slp/src/bchaddr.js:518:15)\n    at decodeAddress (/home/safeuser/SLPDB/node_modules/bchaddrjs-slp/src/bchaddr.js:177:9)\n    at Object.toSlpAddress (/home/safeuser/SLPDB/node_modules/bchaddrjs-slp/src/bchaddr.js:133:17)\n    at Function.Utils.toSlpAddress (/home/safeuser/SLPDB/node_modules/slpjs/lib/utils.js:37:24)\n    at SlpTokenGraph.<anonymous> (/home/safeuser/SLPDB/SlpTokenGraph.js:296:46)\n    at Generator.next (<anonymous>)\n    at /home/safeuser/SLPDB/SlpTokenGraph.js:7:71\n    at new Promise (<anonymous>)\n    at __awaiter (/home/safeuser/SLPDB/SlpTokenGraph.js:3:12)\n    at asyncForEach (/home/safeuser/SLPDB/SlpTokenGraph.js:285:77)' }
trout@p2pvps:~$ 

```

**Comment by jcramer** [comment]
Chris, The issue was caused from a token created with P2PK scriptPubKey type.  I am surprised this hasn't come up sooner, but it should be fixed now with 43bda1b.

If you want to make sure your testnet data is perfect, run this once:
1) `git pull`
2) `npm install` (to resolve a minor/unrelated dependency change)
3) `tsc && node index.js run 1293083`

Otherwise if testnet db doesn't need to be perfect 3) can be just `npm start`.

Let me know how it goes.

**Comment by christroutner** [comment]
Looks like your updates fixed the issue. I'm pretty sure my copy of SLPDB just finished syncing successfully to testnet.
